Corrfunc/utils.py

```diff
@@ -853,6 +853,101 @@ def gridlink_sphere(thetamax,
     else:
         return sphere_grid
 
+
+def convert_to_native_endian(array):
+    '''
+    Returns the supplied array in native endian byte-order.
+    If the array already has native endianness, then the
+    same array is returned.
+
+    Parameters
+    ----------
+    array: np.ndarray
+        The array to convert
+
+    Returns
+    -------
+    new_array: np.ndarray
+        The array in native-endian byte-order.
+
+    Example
+    -------
+    >>> import numpy as np
+    >>> import sys
+    >>> sys_is_le = sys.byteorder == 'little'
+    >>> native_code = sys_is_le and '<' or '>'
+    >>> swapped_code = sys_is_le and '>' or '<'
+    >>> native_dt = np.dtype(native_code + 'i4')
+    >>> swapped_dt = np.dtype(swapped_code + 'i4')
+    >>> arr = np.arange(10, dtype=native_dt)
+    >>> new_arr = convert_to_native_endian(arr)
+    >>> arr is new_arr
+    True
+    >>> arr = np.arange(10, dtype=swapped_dt)
+    >>> new_arr = convert_to_native_endian(arr)
```

Comment:
Will `arr is new_arr` return `False`? If so, may be good to add that in as a check as well

Reply:
It will be `False` by definition of the following check.  If `new_arr` has a different dtype than `arr`, then they can't be the same object!

Reply:
As in the doctest `arr is new_arr == False` is so trivial, that it is not worth including?

Reply:
(Given the amount of stupid mistakes I make on a daily basis, I am always in favor of including higher numbers of checks :) )

Reply:
Since you are the creator of this PR, I will defer to your wishes :)

Reply:
I actually was going to include the check!  I just ran out of time this afternoon to make the edit.

Reply:
Oops. Sorry - I do agree with your assessment anyway. The next check sort of ensures that my suggested check **must** pass. And if that doesn't occur, the bug is probably in numpy or python.